### DISCUSSION TITLE
Support Punycode

### DISCUSSION THREAD

**Discussion by Vitaliy86** [DNSCrypt/dnscrypt-proxy#2017] [Ideas]
Add support Punycode, and in block list

**Comment by d3cim** [comment]
> Add support Punycode

What do you mean exactly? dnscrypt can already manage ASCII / Punycode domains. 

If the Punycode syntax is used to encode internet addresses, each result string is provided with a so-called ACE prefix (short for ASCII-compatible encoding):

ACE prefix: `xn--`

The ACE prefix ensures that domain names containing hyphens are not misinterpreted as international domain names.

This results in the following coding for the IDN müller-büromöbel:

ACE: `xn--mller-brombel-rmb4fg`

>  and in block list

Again, are you asking for block those domains?

you can do this by add a simple rule `xn--*.*` in `blocked-names.txt` file, but I still don't understand why this choice.
